plugins/embed-optimizer/class-embed-optimizer-tag-visitor.php

```diff
@@ -171,12 +173,20 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 				}
 
 				foreach ( $preconnect_hrefs as $preconnect_href ) {
-					$context->link_collection->add_link(
-						array(
-							'rel'  => 'preconnect',
-							'href' => $preconnect_href,
-						)
-					);
+					foreach ( $context->url_metric_group_collection as $group ) {
+						if ( $group->get_element_max_intersection_ratio( $embed_wrapper_xpath ) <= 0 ) {
```

Comment:
Can the intersection ratio ever be less than 0? I don't think so, and [Gemini confirms](https://g.co/gemini/share/d7c2bfb37ad1). Since we're comparing floats, it's probably not safe to check if the return value is exactly equal to 0. How about:

```suggestion
						if ( ! ( $group->get_element_max_intersection_ratio( $embed_wrapper_xpath ) > 0.0 ) ) {
```